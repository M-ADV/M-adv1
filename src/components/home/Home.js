import React from "react";
import { Link, useNavigate } from "react-router-dom";
{/*import Aboutimg from "../../Images/Aboutus.jpg";
import Contactimg from "../../Images/Contactus.jpg";
import Howtoapplyimg from "../../Images/Howtoapplyimg.jpg";*/}

const Home = () => {
  const Navigate = useNavigate();
  const handleLogInOut = () => {
    const userId = localStorage.getItem("id");
    if (userId) {
      localStorage.removeItem("id");
      Navigate("/");
    } else {
      Navigate("/login");
    }
  };

  return (
    <div>
      <div>
        <h1>
          <center className="landingpage">
            <h1 className="h">
              Welcome to <b>M-Advisor</b>
            </h1>
          </center>
        </h1>
        <p className="paragraph text-light">
          {/* Need Loan? then you are right place ðŸ˜‰ check you eligibility and{" "}
          <br />
          apply for the loan. */}
          Planning to take a Mortage Loan? Use this Mortage Loan EMI Calculator
          <br />
          to estimate the EMI amount payable per month to repay the Mortage
          <br />
          Loan. The Mortage Loan EMI Calculator is easy to use. All you need to
          <br />
          select is the Loan amount, Loan Tenure . After selecting the Loan
          <br />
          Amount, and The Loan Tenure, the calculator will estimate the monthly
          <br />
          EMI amount and the total Interest payable till the end of the loan
          tenure.
        </p>
        {!localStorage.getItem("id") && (
          <>
            <button
              className="btn btn-outline-dark home-btn1 bg-dark"
              type="submit"
            >
              <Link className="nav-item nav-link text-light" to="/emicheck">
                Eligibility Checker
              </Link>
            </button>

            <button
              className="btn btn-outline-dark home-btn1 bg-dark"
              type="submit"
            >
              <Link className="nav-item nav-link text-light" to="/Contactus">
                Queries
              </Link>
            </button>
            
          </>
        )}
      </div>
    </div>
  );
};

export default Home;
