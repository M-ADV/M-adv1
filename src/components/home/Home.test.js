import { render, screen } from "@testing-library/react";
import { renderComponent } from "../../utils/test_helpers";
import Home from "./Home";

describe("Home page", () => {
  test("Should Render Intial Page", () => {
    render(renderComponent(<Home />));
    const text1 = screen.getAllByText(/M-Advisior/i);
    expect(text1[0]).toBeInTheDocument();
    const text2 = screen.getAllByText(
      /to estimate the EMI amount payable per month to repay the Mortage/i
    );
    expect(text2[0]).toBeInTheDocument();
  });
  test("should test Intial page buttons", () => {
    render(renderComponent(<Home />));
    expect(
      screen.getByRole("button", { name: "PropertyDetails" })
    ).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "About" })).toBeInTheDocument();
  });
 
});
