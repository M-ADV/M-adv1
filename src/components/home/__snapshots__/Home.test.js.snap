// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Matching the snapshot of the navbar 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        <center
          class="landingpage"
        >
          <h1
            class="h"
          >
            Welcome to 
            <b>
              M-Advisor
            </b>
          </h1>
        </center>
      </h1>
      <p
        class="paragraph text-light"
      >
        Planning to take a Mortage Loan? Use this Mortage Loan EMI Calculator
        <br />
        to estimate the EMI amount payable per month to repay the Mortage
        <br />
        Loan. The Mortage Loan EMI Calculator is easy to use. All you need to
        <br />
        select is the Loan amount, Loan Tenure . After selecting the Loan
        <br />
        Amount, and The Loan Tenure, the calculator will estimate the monthly
        <br />
        EMI amount and the total Interest payable till the end of the loan tenure.
      </p>
      <button
        class="btn btn-outline-dark home-btn1 bg-dark"
        type="submit"
      >
        <a
          class="nav-item nav-link text-light"
          href="/emicheck"
        >
          Eligibility Checker
        </a>
      </button>
      <button
        class="btn btn-outline-dark home-btn1 bg-dark"
        type="submit"
      >
        <a
          class="nav-item nav-link text-light"
          href="/Contactus"
        >
          Queries
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
